app/app.go

```diff
@@ -299,7 +346,14 @@ func (app *GaiaApp) InitChainer(ctx sdk.Context, req abci.RequestInitChain) abci
 	var genesisState simapp.GenesisState
 	app.cdc.MustUnmarshalJSON(req.AppStateBytes, &genesisState)
 
-	return app.mm.InitGenesis(ctx, app.cdc, genesisState)
+	res := app.mm.InitGenesis(ctx, app.cdc, genesisState)
+
+	// Set Historical infos in InitChain to ignore genesis params
+	stakingParams := staking.DefaultParams()
+	stakingParams.HistoricalEntries = 1000
+	app.stakingKeeper.SetParams(ctx, stakingParams)
```

Comment:
shouldn't we set 1000 in the default params instead then?

Reply:
That would be ideal!

Reply:
@AdityaSripal what do you think about changing the default? We can add a comment on the app.go and simapp.go that users who do not want to store historical entries don't need to add them to the application's `SetOrderBeginBlockers`

Reply:
This is a change that will need to be made in the SDK. I'm going to leave this here since it is required for IBC to work.